core/node/eth_sender/src/aggregator.rs

```diff
@@ -44,65 +44,63 @@ impl Aggregator {
-                }),
-            ],
-            proof_criteria: vec![
-                Box::from(NumberCriterion {
-                    op: AggregatedActionType::PublishProofOnchain,
-                    limit: *config.aggregated_proof_sizes.iter().max().unwrap() as u32,
-                }),
-                Box::from(GasCriterion::new(
-                    AggregatedActionType::PublishProofOnchain,
-                    config.max_aggregated_tx_gas,
-                )),
-                Box::from(TimestampDeadlineCriterion {
-                    op: AggregatedActionType::PublishProofOnchain,
-                    deadline_seconds: config.aggregated_block_prove_deadline,
-                    // Currently, we can't use this functionality for proof criterion
-                    // since we don't send dummy and real proofs in the same range,
-                    // so even small ranges must be closed.
-                    max_allowed_lag: None,
-                }),
-            ],
-            execute_criteria: vec![
+        let operate_4844_mode =
+            custom_commit_sender_addr.is_some() && !settlement_mode.is_gateway();
+
+        // We do not have a reliable lower bound for gas needed to execute batches on gateway so we do not aggregate.
+        let execute_criteria: Vec<Box<dyn L1BatchPublishCriterion>> = if settlement_mode
+            .is_gateway()
+        {
+            if config.max_aggregated_blocks_to_execute > 1 {
+                tracing::warn!(
+                    "config.max_aggregated_blocks_to_execute is set to {} but \
+                    aggregator does not support aggregating execute operations when settling on gateway",
+                    config.max_aggregated_blocks_to_execute
+                );
+            }
+
+            vec![Box::from(NumberCriterion {
+                op: AggregatedActionType::Execute,
+                limit: 1,
+            })]
+        } else {
+            vec![
                 Box::from(NumberCriterion {
                     op: AggregatedActionType::Execute,
                     limit: config.max_aggregated_blocks_to_execute,
                 }),
-                Box::from(GasCriterion::new(
-                    AggregatedActionType::Execute,
-                    config.max_aggregated_tx_gas,
-                )),
                 Box::from(TimestampDeadlineCriterion {
                     op: AggregatedActionType::Execute,
                     deadline_seconds: config.aggregated_block_execute_deadline,
                     max_allowed_lag: Some(config.timestamp_criteria_max_allowed_lag),
                 }),
-            ],
+                Box::from(ExecuteGasCriterion::new(config.max_aggregated_tx_gas)),
+            ]
+        };
+
+        if config.max_aggregated_blocks_to_commit > 1 {
+            tracing::warn!(
+                "config.max_aggregated_blocks_to_commit is set to {} but \
+                aggregator does not support aggregating commit operations anymore",
```

Comment:
Discussed in private, it may make sense to support these for L1 especially in the short term